tests/unit/models/gpflow/test_models.py

```diff
@@ -643,3 +643,92 @@ def test_gpflow_predictor_get_observation_noise_raises_for_likelihood_with_varia
+
+    pred_mean7, pred_var7 = model7.predict(query_points)
+    pred_mean5, pred_var5 = model5.conditional_predict_f(query_points, additional_data)
+
+    predy_mean7, predy_var7 = model7.predict_y(query_points)
+    predy_mean5, predy_var5 = model5.conditional_predict_y(query_points, additional_data)
+
+    np.testing.assert_allclose(tf.transpose(tf.linalg.diag_part(predj_cov5)), pred_var5, atol=1e-5)
+    np.testing.assert_allclose(predj_mean5, pred_mean5, atol=1e-5)
+    np.testing.assert_allclose(predj_mean5, predj_mean7, atol=1e-5)
+    np.testing.assert_allclose(pred_mean7, pred_mean5, atol=1e-5)
+    np.testing.assert_allclose(pred_var7, pred_var5, atol=1e-5)
+    np.testing.assert_allclose(predj_cov7, predj_cov5, atol=1e-5)
+    np.testing.assert_allclose(predy_mean7, predy_mean5, atol=1e-5)
+    np.testing.assert_allclose(predy_var7, predy_var5, atol=1e-5)
+
+
+def test_gaussian_process_regression_conditional_predict_equations_broadcast() -> None:
+    x = gpflow.utilities.to_default_float(
+        tf.constant(np.arange(1, 24).reshape(-1, 1) / 8.0)
+    )  # shape: [23, 1]
+    y = fnc_2sin_x_over_3(x)
+
+    model5 = GaussianProcessRegression(gpr_model(x[:5, :], y[:5, :]))
+
+    additional_data = Dataset(tf.reshape(x[5:, :], [3, 6, -1]), tf.reshape(y[5:, :], [3, 6, -1]))
+
+    query_points = tf.concat([0.5 * x, 2.0 * x], 0)  # shape: [46, 1]
+
+    predj_mean5, predj_cov5 = model5.conditional_predict_joint(query_points, additional_data)
+    pred_mean5, pred_var5 = model5.conditional_predict_f(query_points, additional_data)
+    predy_mean5, predy_var5 = model5.conditional_predict_y(query_points, additional_data)
+
+    for i in range(3):
+        xi = tf.concat([x[:5, :], additional_data.query_points[i, ...]], axis=0)
+        yi = tf.concat([y[:5, :], additional_data.observations[i, ...]], axis=0)
+
+        modeli = GaussianProcessRegression(gpr_model(xi, yi))
+        predj_meani, predj_covi = modeli.predict_joint(query_points)
+        pred_meani, pred_vari = modeli.predict(query_points)
+        predy_meani, predy_vari = modeli.predict_y(query_points)
+
+        np.testing.assert_allclose(predj_mean5[i, ...], predj_meani, atol=1e-5)
+        np.testing.assert_allclose(pred_meani, pred_mean5[i, ...], atol=1e-5)
+        np.testing.assert_allclose(pred_vari, pred_var5[i, ...], atol=1e-5)
+        np.testing.assert_allclose(predj_covi, predj_cov5[i, ...], atol=1e-5)
+        np.testing.assert_allclose(predy_vari, predy_var5[i, ...], atol=1e-5)
+        np.testing.assert_allclose(predy_vari, predy_var5[i, ...], atol=1e-5)
+
+
+def test_gaussian_process_regression_conditional_predict_f_sample() -> None:
+    x = gpflow.utilities.to_default_float(
+        tf.constant(np.arange(1, 24).reshape(-1, 1) / 8.0)
+    )  # shape: [23, 1]
+    y = fnc_2sin_x_over_3(x)
+
+    model5 = GaussianProcessRegression(gpr_model(x[:5, :], y[:5, :]))
+    additional_data = Dataset(tf.reshape(x[5:, :], [3, 6, -1]), tf.reshape(y[5:, :], [3, 6, -1]))
```

Comment:
why this reshaping?

Reply:
emulating multiple batch - when you optimise over a batch, our optimiser will start by evaluating a set of random batches before running BFGS.